### PR TITLE
[Bugfix] Fix a small device bug

### PR DESCRIPTION
**Pull request by szhengac**
Initialize all-gather tensor on gpu.

## Checklist ##

- [x] PR's title starts with a category (e.g. [Bugfix], [Model], [Tutorial], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### CODE REVIEW

**Review by comaniac — APPROVED** [review]

**Comment by chhzh123** [comment]
This line also uses `.to(device)`. Maybe we can also change it?
https://github.com/awslabs/slapo/blob/main/slapo/framework_dialect/deepspeed/pipeline.py#L320

**Comment by szhengac** [comment]
> This line also uses `.to(device)`. Maybe we can also change it? https://github.com/awslabs/slapo/blob/main/slapo/framework_dialect/deepspeed/pipeline.py#L320

Nice catch!

File: slapo/schedule.py [hunk 1]
```diff
@@ -261,10 +268,10 @@ def _shard(name, tensor):
                     self.metadata.tie_weights[param] = new_param
             else:
                 new_param = nn.Parameter(new_tensor)
-
+            # Tag param with model parallel attribute, used for grad clipping
+            _set_model_parallel_attribute(new_param, TENSOR_MODEL_PARALLEL, True)
```
Comment (comaniac) [inline slapo/schedule.py:272]: We should use the same way to set `orig_shape`.

**Comment by comaniac** [comment]
Thanks @szhengac @chhzh123

**Pull request merged**